### PR TITLE
My/document

### PR DESCRIPTION
**Pull request by myyong**
1. Docstrings added for main.py and settings.py.
2. Sphinx documents added so that docstrings can be formatted and published to readthedocs.

### CODE REVIEW

File: docs/temp/automatethis.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# remember to place requirements in docs/
```
Comment (Iain-S) [inline docs/temp/automatethis.txt:1]: Is this directory meant to be under version control?

File: sqlsynthgen/main.py [hunk 2]
```diff
@@ -28,7 +39,31 @@ def create_data(
...
+    by file containing Python classes and its attributes.
+
+    Takes as input sqlsynthgen output as represented by Python
+    classes, its attributes and methods for generating values
+    for those attributes.
+
+    Final input is the number of rows required.
+
+    Example:
+        $ python sqlsynthgen/main.py create-data example_orm.py expected_ssg.py 100
```
Comment (Iain-S) [inline sqlsynthgen/main.py:57]: The CLI examples can all be:

```suggestion
        $ sqlsynthgen create-data example_orm.py expected_ssg.py 100
```

without the `python sqlsynthgen/main.py`

File: sqlsynthgen/main.py [hunk 3]
```diff
@@ -38,22 +73,65 @@ def create_data(
...
 @app.command()
 def create_tables(orm_file: str = typer.Argument(...)) -> None:
-    """Create tables using the SQLAlchemy file."""
+    """Create schema from Python classes
+
+    This CLI command creates Postgresql schema using object relational model
+    declared as Python tables. (eg.)
+
+    Example:
+        $ python sqlsynthgen/main.py create-tables example_orm.py
```
Comment (Iain-S) [inline sqlsynthgen/main.py:82]: ditto

File: sqlsynthgen/main.py [hunk 4]
```diff
@@ -38,22 +73,65 @@ def create_data(
...
 @app.command()
 def make_generators(orm_file: str = typer.Argument(...)) -> None:
-    """Make a SQLSynthGen file of generator classes."""
+    """Make a SQLSynthGen file of generator classes.
+
+    This CLI command takes an object relation model output by sqlcodegen and
+    returns a set of synthetic data generators for each attribute
+
+    Example:
+        $ python sqlsynthgen/main.py make-generators example_orm.py
```
Comment (Iain-S) [inline sqlsynthgen/main.py:103]: ditto

File: sqlsynthgen/main.py [hunk 5]
```diff
@@ -38,22 +73,65 @@ def create_data(
...
 def make_tables() -> None:
-    """Make a SQLAlchemy file of Table classes."""
+    """Make a SQLAlchemy file of Table classes.
+
+    This CLI command deploys sqlacodegen to discover a
+    schema structure, and generates a object relational model declared
+    as Python classes.
+
+    Example:
+        $ python sqlsynthgen/main.py make_tables
```
Comment (Iain-S) [inline sqlsynthgen/main.py:126]: ditto

File: sqlsynthgen/main.py [hunk 6]
```diff
@@ -73,7 +137,22 @@ def make_generators(
...
+
+    Args:
+        None
```
Comment (Iain-S) [inline sqlsynthgen/main.py:150]: Can we do without the `Args:` section if there are none?

File: sqlsynthgen/main.py [hunk 7]
```diff
@@ -73,7 +137,22 @@ def make_generators(
...
+
+    Returns:
+        None
```
Comment (Iain-S) [inline sqlsynthgen/main.py:153]: Is it OK to leave out the `Returns:` section if there is no return value (though, confusingly, functions without explicit `return` statements do return `None`)

File: pyproject.toml [hunk 8]
```diff
@@ -15,14 +15,16 @@ sqlalchemy-utils = "^0.38.3"
...
+sqlalchemy = "^1.4"
 
 [tool.poetry.group.dev.dependencies]
 black = "^22.10.0"
 isort = "^5.10.1"
 pylint = "^2.15.8"
 mypy = "^0.991"
 types-pyyaml = "^6.0.12.4"
+python-docs-theme = "^2022.1"
+sphinx-rtd-theme = "^1.1.1"
```
Comment (Iain-S) [inline pyproject.toml:27]: Are these needed given that there is also a `docs/requirements.txt`?

Comment (Iain-S) [reply pyproject.toml:27]: Or, rather, can we remove `docs/requirements.txt` if we have the requirements here?

File: sqlsynthgen/main.py (hunk 6 above)
Comment (myyong) [reply sqlsynthgen/main.py:150]: It looks look more readable without. I agree with removing it.

File: sqlsynthgen/main.py (hunk 7 above)
Comment (myyong) [reply sqlsynthgen/main.py:153]: I think if we remove the cases where args is `None`, we can do the same for returns.

File: pyproject.toml (hunk 8 above)
Comment (myyong) [reply pyproject.toml:27]: The requirements.txt file is used by readthedocs.

Though we do not explicitly need it in dev.dependencies, it feels like having the ability to work on documentation should be part of the package should someone choose to fork and take it on. Happy to be corrected on this.

**Review by Iain-S — APPROVED** [review]
LGTM

**Pull request merged**